common/src/main/kotlin/entity/DiscordGuild.kt

```diff
@@ -557,4 +574,28 @@ data class DiscordWelcomeScreen(
     val description: String?,
     @SerialName("welcome_channels")
     val welcomeChannels: List<DiscordWelcomeScreenChannel>
-)
\ No newline at end of file
+)
+
+@Serializable
+data class DiscordMembershipScreening(
+    val version: String,
+    @SerialName("form_fields")
+    val formFields: List<FormField>,
+    val description: Optional<String> = Optional.Missing()
+) {
+
+    @Serializable
+    data class FormField(
+        @SerialName("field_type")
+        val type: Type,
+        val label: String,
+        val values: Optional<List<String>> = Optional.Missing(),
+        val required: Boolean
+    ) {
+        @Serializable
+        enum class Type {
```

Comment:
Use a more expresive name than `Type`, maybe `FormFieldType` is a better. name
Also use objects so we can have `Unknown` type in there